apps/src/codebridge/FileBrowser/prompts/openMoveFilePrompt.tsx

```diff
@@ -32,48 +34,41 @@ export const openMoveFilePrompt = async ({
   sendCodebridgeAnalyticsEvent,
 }: OpenMoveFilePromptArgsType) => {
   const file = projectFiles[fileId];
-  const results = await dialogControl?.showDialog({
-    type: DialogType.GenericPrompt,
-    title: codebridgeI18n.moveFilePrompt(),
-    placeholder: codebridgeI18n.rootFolder(),
-    requiresPrompt: false,
 
-    validateInput: (destinationFolderName: string) => {
-      try {
-        const folderId = findFolder(destinationFolderName.split('/'), {
-          folders: Object.values(projectFolders),
-          required: true,
-        });
+  // iterate over all the folders in the project AND the default folder, which isn't actually in the list.
+  const possibleDestinationFolders: GenericDropdownProps['items'] = [
+    {id: DEFAULT_FOLDER_ID},
+    ...Object.values(projectFolders),
+  ]
+    .filter(
+      f =>
+        !Boolean(
+          validateFileName({
+            fileName: file.name,
+            folderId: f.id,
+            projectFiles,
+            isStartMode,
+            validationFile,
+          })
+        )
+    )
+    .map(f => ({value: f.id, text: getFolderPath(f.id, projectFolders)}))
+    .sort((a, b) => a.text.localeCompare(b.text));
 
-        return validateFileName({
-          fileName: file.name,
-          folderId,
-          projectFiles,
-          isStartMode,
-          validationFile,
-        });
-      } catch (e) {
-        return getErrorMessage(e);
-      }
-    },
+  const results = await dialogControl?.showDialog({
+    type: DialogType.GenericDropdown,
+    title: codebridgeI18n.moveFilePrompt(),
+    selectedValue: possibleDestinationFolders[0].value,
```

Comment:
could there theoretically be an empty list? If you had no folders (i.e. just the root) there would be no valid folders to move to right?

Reply:
Oof. Good catch! You're completely right and it wasn't handling it.

The folder case is rare - afaik it only appears if you have a single folder in the project. The only possible places to move that would be to the root (where it already is) or into itself. Neither of those should be allowed, so it filters to an empty list.

The file case is still unlikely, there are more opportunities for it - if you have a file with the same name in every single available directory, it'd manifest.

So.

I refactored out the inlined functions in `openMoveFilePrompt` and `openMoveFolderPrompt` into `getPossibleDestinationFoldersForFile` and `getPossibleDestinationFoldersForFolder` respectively.

it might be good to further add additional wrappers - `canMoveFile` and `canMoveFolder`, but I'm gonna circle back to that since I think there may be more refactoring here.

Anyway, with those two functions pulled out, I wrapped the `Move Folder` `PopUpOption` in the `FileBrowser` to only display if it has a folder it can move to, as well as extended the `condition` on the `moveFile` option in `FileRow` to confirm whether there's anywhere it can actually move _to_.

In either case, if it finds nowhere to move the file/folder to, it removes the `Move` option from the kebab menu.

I want to circle back to this in a future PR (since I'm planning on more work in this area around retooling the folder list to use the `FileRow`) to confirm that this is the proper way we want to convey this concept. Other ideas would include:
- Don't display the move prompt (how it currently works). This may be confusing because...why isn't it there?
- Display the move prompt, but disable it. This shows that there _might_ be an option, but it may be confusing why it's disabled. Maybe we clear that up with a tooltip? This is probably bad UX - we should avoid displaying options the user can't select.
- Keep the move prompt, but have it pop up an error that there is nowhere to move to. This is very bad UX, we shouldn't let the user select an option which will do nothing but fail.

Reply:
It seems reasonable to me to hide move if there's nothing to move to. It will be a pretty common case actually, most of the start code for data science has only the root folder.

Reply:
We should still guard against the empty case here; I know we are checking elsewhere and not calling the function if there are no possible destination folders, but it's good to avoid throwing an exception since it would crash the page.

Reply:
Is there another case to guard against?

That comment there was outlining the mods that I had put in to cover it after your suggestion, so I think we're protected now.

Reply:
I did update the prompts themselves so if somehow the function were to be called w/o destinations, it'll just be a no-op w/o attempting to open the dialog.

Reply:
The case I'm thinking about it if `possibleDestinationFolders` is empty, and we try to access index 0.